orca_detector/noise_collector.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: future_fstrings -*-
+
+"""
+Script to collect random samples of noise from live streams.
+
+W251 (Summer 2019) - Spyros Garyfallos, Ram Iyer, Mike Winton
+"""
+
+import m3u8
+import numpy as np
+import orca_params
+import os
+import random
+import time
+import uuid
+
+# Dictionary of streams we're recording from
+streams = {'OrcasoundLab': 'https://s3-us-west-2.amazonaws.com/streaming-orcasound-net/rpi_orcasound_lab/hls/1562344334/live.m3u8'}
+
+def _save_audio(audio_url, output_path):
+    """
+    Uses ffmpeg (via CLI) to retrieve an audio segment from audio_url
+    and save it to the local output_path.
+    """
+
+    # generate random file names in case stream reuses its names
+    file_name = f"{uuid.uuid4().hex}.wav"
+    output_file = os.path.join(output_path, file_name)
+    print(f'Saving audio segment: {audio_url}')
+    print(f' to {output_file}')
+    
+    ffmpeg_cli = f'ffmpeg -y -i {audio_url} {output_file} -loglevel warning'
```

Comment:
I was having a hard time finding a Python wrapper module for ffmpeg that was well-documented and not buggy, so I resorted to having Python call it from the CLI.

Reply:
One other thing to investigate is the m3u8 links, I think these links change over time (the id part in the URI). If this is the case we might have to scrape the links from the html page (BeautifulSoup and get the *.m3u8 links)

Reply:
Here's the script I was planning to use for our inference. Create 1 second file segments from the stream listener app (with a timestamp filename), have a tumbling window of the last 5 seconds an run inference on the inference app.

https://superuser.com/questions/999400/how-to-use-ffmpeg-to-extract-live-stream-into-a-sequence-of-mp4

Reply:
The m3u8 playlists include a bunch of 10 second samples, so I don't need `-t`.  Each sample is exactly the same size.

I spent a while watching the m3u8 file, and it doesn't seem to continuously update, so it would be possible to retrieve the file multiple times and get the same list of files.  So, I think random is the way to go.

Re: m3u8 links, yeah I share that concern (that they'll be transient, too).  For now, I'm just going to keep it running, and if I run into errors I'll do an update (and follow-up PR) at that point.  

I decided not to try to pull samples of explicit length (e.g. 1 sec), but rather treat these samples just like all of the other species' samples in terms of quantization when crawling the directory tree.